scripts/ai/AIDriveStrategyUnloadCombine.lua

```diff
@@ -1583,29 +1588,33 @@ end
 ------------------------------------------------------------------------------------------------------------------------
 -- Set up a course to move out of the way of a blocking vehicle
 ------------------------------------------------------------------------------------------------------------------------
-function AIDriveStrategyUnloadCombine:createMoveAwayCourse(blockingVehicle)
-    local trailer = AIUtil.getImplementOrVehicleWithSpecialization(self.vehicle, Trailer)
-    -- if we look straight left or right out of the window, is blockingVehicle in front of us or behind us?
-    -- if in front, move back, if behind, move forward
-    -- but since we have a trailer, don't use the tractor's direction node directly, instead, a point behind it
-    -- about the half length of the rig.
-    local _, frontMarkerOffset = Markers.getFrontMarkerNode(self.vehicle)
-    local _, backMarkerOffset = Markers.getBackMarkerNode(self.vehicle)
-    local _, _, dz = localToLocal(blockingVehicle.rootNode, self.vehicle:getAIDirectionNode(), 0, 0, 0)
-    if dz > (frontMarkerOffset + backMarkerOffset) / 2 then
-        self:debug('%s is in front, moving back (dz %.1f, front %.1f, back %.1f)', CpUtil.getName(blockingVehicle),
-                dz, frontMarkerOffset, backMarkerOffset)
+function AIDriveStrategyUnloadCombine:createMoveAwayCourse(blockingVehicle, isAheadOfUs, trailer)
+    if isAheadOfUs then
+        self:debug('%s is in front, moving back', CpUtil.getName(blockingVehicle))
         -- blocking vehicle in front of us, move back, calculate course from the trailer's root node
         return Course.createFromNode(self.vehicle, trailer.rootNode, 0, -2, -27, -5, true)
     else
         -- blocking vehicle behind, move forward
-        self:debug('%s is behind us, moving forward (dz: %.1f, front %.1f, back %.1f)', CpUtil.getName(blockingVehicle),
-                dz, frontMarkerOffset, backMarkerOffset)
+        local _, frontMarkerOffset = Markers.getFrontMarkerNode(self.vehicle)
+        self:debug('%s is behind us, moving forward', CpUtil.getName(blockingVehicle))
         return Course.createFromNode(self.vehicle, self.vehicle:getAIDirectionNode(), 0,
                 frontMarkerOffset, frontMarkerOffset + self.maxDistanceWhenMovingOutOfWay, 5, false)
     end
 end
 
+------------------------------------------------------------------------------------------------------------------------
+-- Is the blocking vehicle in front of us?
+------------------------------------------------------------------------------------------------------------------------
+function AIDriveStrategyUnloadCombine:isBlockingVehicleAheadOfUs(blockingVehicle)
```

Comment:
Maybe we could place this function in a util container, so we can reuse this in another strategy?